front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -166,18 +169,33 @@ export function AssistantInputBar({
       ...new Set(rawMentions.map((mention) => mention.id)),
     ].map((id) => ({ configurationId: id }));
 
-    onSubmit(
-      text,
-      mentions,
-      fileUploaderService.getFileBlobs().map((cf) => {
-        return {
-          title: cf.filename,
-          fileId: cf.fileId,
-        };
-      })
-    );
-    resetEditorText();
-    fileUploaderService.resetUpload();
+    let delayedLoading: number = 0;
+    // When we are creating a new conversation, we will disable the input bar, show a loading spinner and in case of error, re-enable the input bar
+    if (!conversationId) {
+      delayedLoading = _.delay(() => setLoading(true), 250);
```

Comment:
Why not using setTimeout in a `useEffect`. You need to clear the timeout, which IIRC `_.delay` does not do.

Reply:
I don't have access to setLoading outside of this method (it's a parameter of `onEnterKeyDown`).
I clear the timeout below btw.
Will use setTimetout / clearTimeout to make it clearer (i started with _.debounce)